pkg/storage/snapshot/restore.go

```diff
@@ -454,6 +457,56 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	return t.reconcileSpec(restoredVM)
 }
 
+func (t *vmRestoreTarget) updateVMRestoreRestores() (bool, error) {
+
+	content, err := t.controller.getSnapshotContent(t.vmRestore)
+	if err != nil {
+		return false, err
+	}
+
+	snapshotVM := content.Spec.Source.VirtualMachine
+	if snapshotVM == nil {
+		return false, fmt.Errorf("unexpected snapshot source")
+	}
+	var restores = make([]snapshotv1.VolumeRestore, len(t.vmRestore.Status.Restores))
+	for i, t := range t.vmRestore.Status.Restores {
+		t.DeepCopyInto(&restores[i])
+	}
+	for k := range restores {
+		restore := &restores[k]
+		for _, volume := range snapshotVM.Spec.Template.Spec.Volumes {
+			if volume.Name != restore.VolumeName {
+				continue
+			}
+			if volume.DataVolume != nil {
+				templateIndex := findDVTemplateIndex(volume.DataVolume.Name, snapshotVM)
```

Comment:
can you get rid of this call by changing line 477 to be:

```golang
for templateIndex, volume := range snapshotVM.Spec.Template.Spec.Volumes {
```

Reply:
its not the same.. a volume in spec.volumes is not necessarily from a dvtemplate..